### PR TITLE
Rename `db` log category to `walletdb` (like `coindb`)

### PR DESCRIPTION
**Pull request by laanwj**
Rename the `db` log category to `walletdb` (in the style of, and to distinguish from `coindb`). Deprecate (but still accept) '-debug=db'.

Second commit is a scripted commit that changes the enum item name.

### CODE REVIEW

**Comment by practicalswift** [comment]
ACK 97330653b49c4b7d5e7646a28dc14906a574a005 -- diff looks correct and disambiguation is good

**Comment by MarcoFalke** [comment]
ACK 97330653b49c4b7d5e7646a28dc14906a574a005

needs release notes

**Comment by laanwj** [comment]
Rebased and added release note.

File: doc/release-notes-17410.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Command-line options
+--------------------
+
+- The `-debug=db` logging category has been renamed to `-debug=walletdb`, to distinguish it from `coindb`.
```
Comment (MarcoFalke) [inline doc/release-notes-17410.md]:
```suggestion
- The `-debug=db` logging category has been renamed to `-debug=walletdb`, to distinguish it from `coindb`.
   `-debug=db` has been deprecated and will be removed in the next major release.
```

Comment (laanwj) [reply doc/release-notes-17410.md]: Thanks, will add that.
Was trying to keep it succinct, this is not worth making the release notes much longer over. But that's good to mention.

**Comment by jonasschnelli** [comment]
utACK 1a0007218f15f85e2c7952bbdb4a84ec2351bfe2

**Review by hebasto — APPROVED** [review]
ACK e2c03c1156a1a8cb2c04c180f2ddbd3535126a46, tested on Linux Mint 19.2:

```
$ ./src/bitcoind -debug=db
...
2019-11-09T12:21:25Z [init] Warning: logging category 'db' is deprecated, use 'walletdb' instead
...
```

```
$ ./src/bitcoin-cli logging | grep true
  "walletdb": true,
```

**Pull request merged**